tools/chplcheck/src/driver.py

```diff
@@ -135,9 +142,19 @@ def advanced_rule(self, func):
 
         The name of the decorated function is used as the name of the rule.
         """
-
-        self.AdvancedRules.append((func.__name__, func))
-        return func
+        def decorator_advanced_rule(func):
+            self.AdvancedRules.append((func.__name__, func))
+            if not default:
+                self.SilencedRules.append(func.__name__)
+
+            @functools.wraps(func)
+            def wrapper_advanced_rule(*args, **kwargs):
+                return func(*args, **kwargs)
+            return wrapper_advanced_rule
+        if _func is None:
+            return decorator_advanced_rule
+        else:
+            return decorator_advanced_rule(_func)
```

Comment:
I think this is because `advanced_rule` and `advanced_rule(...)` means different things. I suggest you add a comment about this.